client/base_client.go

```diff
@@ -99,18 +106,26 @@ func WithMaxErrorRetry(count int) ClientOption {
 	}
 }
 
+// WithMaxTSOBatchWaitInterval configures the client max TSO batch wait interval.
+func WithMaxTSOBatchWaitInterval(maxTSOBatchWaitInterval time.Duration) ClientOption {
+	return func(c *baseClient) {
+		c.maxTSOBatchWaitInterval = maxTSOBatchWaitInterval
+	}
+}
+
 // newBaseClient returns a new baseClient.
 func newBaseClient(ctx context.Context, urls []string, security SecurityOption, opts ...ClientOption) (*baseClient, error) {
 	ctx1, cancel := context.WithCancel(ctx)
 	c := &baseClient{
-		urls:                 urls,
-		checkLeaderCh:        make(chan struct{}, 1),
-		checkTSODispatcherCh: make(chan struct{}, 1),
-		ctx:                  ctx1,
-		cancel:               cancel,
-		security:             security,
-		timeout:              defaultPDTimeout,
-		maxRetryTimes:        maxInitClusterRetries,
+		urls:                    urls,
+		checkLeaderCh:           make(chan struct{}, 1),
+		checkTSODispatcherCh:    make(chan struct{}, 1),
+		ctx:                     ctx1,
+		cancel:                  cancel,
+		security:                security,
+		timeout:                 defaultPDTimeout,
+		maxRetryTimes:           maxInitClusterRetries,
+		maxTSOBatchWaitInterval: defaultMaxTSOBatchInterval,
```

Comment:
I think it should disable by default, the latency is slightly increased if enabled.